node/sovereignNodeFactory_test.go

```diff
@@ -25,3 +25,17 @@ func TestSovereignNodeFactory_CreateNewNode(t *testing.T) {
 	require.NotNil(t, sn)
 	require.Equal(t, "*node.sovereignNode", fmt.Sprintf("%T", sn))
 }
+
+func TestSovereignNodeFactory_CreateNewNodeFail(t *testing.T) {
+	t.Parallel()
+
+	sovereignNodeFactory := node.NewSovereignNodeFactory()
+
+	options := []node.Option{
+		node.WithStatusCoreComponents(nil),
+	}
+
+	sn, err := sovereignNodeFactory.CreateNewNode(options...)
+	require.Equal(t, "error applying option: nil status core components", err.Error())
```

Comment:
We should better check that the error contains the `ErrNilStatusCoreComp...` here instead of checking for string.
Reason being that if somebody changes the error message your test would fail